plugins/autowarmth.koplugin/main.lua

```diff
@@ -150,12 +150,20 @@ function AutoWarmth:scheduleMidnightUpdate()
         end
     end
 
+    -- unpack only goes until the first nil value, but we need the whole array
+    function fullunpack (t, i, max)
+      i = i or 1
+      while t[i] ~= nil or (max and i <= max) do
+        return t[i], fullunpack(t, i + 1, max)
+      end
+    end
```

Comment:
c.f., https://github.com/koreader/koreader/blob/71cea5a9f6cfb9961c18346cdee76c78dbc04dcf/frontend/ui/event.lua#L33-L36

&&

https://github.com/koreader/koreader/blob/71cea5a9f6cfb9961c18346cdee76c78dbc04dcf/frontend/ui/widget/eventlistener.lua#L32

Reply:
Ups, I reinvented the wheel. Thank you.